File: tests/gates.ts
// nullgate "-"
function H (qubit) {
  if (typeof qubit === "number") {
    return `qc.h(${qubit})`;
  }
  if (Array.isArray(qubit)) {
    return `qc.h(${JSON.stringify(qubit)})`;
  };
}

export default {
  "-": (_) => ``,
  H,
  CX: (c, t) => `qc.cx(${c}, ${t})`,
  CCX: (c1, c2, t) => `qc.ccx(${c1}, ${c2}, ${t})`,

  X: (q) => `qc.x(${q})`,
  Y: (q) => `qc.y(${q})`,
  Z: (q) => `qc.z(${q})`,

  RX: (q, rad) => `qc.rx(${parseFloat(rad)}, ${q})`,
  RY: (q, rad) => `qc.ry(${parseFloat(rad)}, ${q})`,
  RZ: (q, rad) => `qc.rz(${parseFloat(rad)}, ${q})`
}

File: tests/gates.py
def H(qubit):
  if isinstance(qubit, int):
    return f'qc.h({qubit})'
  if isinstance(qubit, list):
    return f'qc.h({json.dumps(qubit)})'

def X(q):
  return f'qc.x({q})'

def Y(q):
  return f'qc.y({q})'

def Z(q):
  return f'qc.z({q})'

def RX(q, rad):
  return f'qc.rx({float(rad)}, {q})'

def RY(q, rad):
  return f'qc.ry({float(rad)}, {q})'

def RZ(q, rad):
  return f'qc.rz({float(rad)}, {q})'

def CX(c, t):
  return f'qc.cx({c}, {t})'

def CCX(c1, c2, t):
  return f'qc.ccx({c1}, {c2}, {t})'

# Define a nullgate function
def nullgate(_):
  return ''